pkg/spoke/addon/registration_controller.go

```diff
@@ -0,0 +1,232 @@
+package addon
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+	certificatesinformers "k8s.io/client-go/informers/certificates/v1beta1"
+	"k8s.io/client-go/kubernetes"
+	csrclient "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+	"k8s.io/klog/v2"
+
+	addoninformerv1alpha1 "github.com/open-cluster-management/api/client/addon/informers/externalversions/addon/v1alpha1"
+	addonlisterv1alpha1 "github.com/open-cluster-management/api/client/addon/listers/addon/v1alpha1"
+	"github.com/open-cluster-management/registration/pkg/clientcert"
+)
+
+// addOnRegistrationController monitors ManagedClusterAddOn crs on hub and starts addOn registration
```

Comment:
enumerate the actual controllers you're starting. The first one is a clientcert.NewClientCertificateController.

Reply:
Added in comments.